### PR TITLE
Adjust cache change endpoint request method

### PR DESCRIPTION
**Pull request by Felixim0**

### CODE REVIEW

File: app/job_role_utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    self.lock = asyncio.Lock()
+
+  async def fetch(self, request):
+    client = request.app['client']
+    if self.cache is None or (time.time() - self.timestamp) > self.lifetime:
+      async with self.lock:
+        url = FSDR_URL + f'/jobRoles/allJobRoleShorts/distinct'
+        auth = BasicAuth(FSDR_USER, FSDR_PASS)
+        async with client.get(url, auth=auth) as resp:
+          self.cache = await resp.json()
```
Comment (SammyPoot) [inline app/job_role_utils.py]: Using a local variable for the first two stages of this computation will be slightly better. `cache` would do.

File: app/microserviceshandler.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ async def get(self, request):
 
     if 'clear' in microservice_name:
       microservice_name = microservice_name.replace('clear', '')
-      await clear_stored_search_criteria(session, microservice_name)
+      await clear_stored_search_criteria(session, request, microservice_name)
```
Comment (SammyPoot) [inline app/microserviceshandler.py]: If we are passing `request`, we do not need to pass `session`

File: app/microserviceshandler.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ async def post(self, request):
...
```
Comment (SammyPoot) [inline app/microserviceshandler.py]: If we are passing `request`, we do not need to pass `session`

File: app/searchcriteria.py [hunk 4]
```diff
@@ -42,7 +42,7 @@ async def store_search_criteria(request, search_criteria, fields_to_load=[]):
       session[atribute] = search_criteria.get(atribute)
 
 
-async def clear_stored_search_criteria(session, microservice_name=''):
+async def clear_stored_search_criteria(session, request, microservice_name=''):
```
Comment (SammyPoot) [inline app/searchcriteria.py]: If we are passing `request`, we do not need to pass `session`

File: app/searchhandler.py [hunk 5]
```diff
@@ -38,7 +38,7 @@ async def get(self, request):
 
     await saml.ensure_logged_in(request)
 
-    await clear_stored_search_criteria(session)
+    await clear_stored_search_criteria(session, request)
```
Comment (SammyPoot) [inline app/searchhandler.py]: If we are passing `request`, we do not need to pass `session`

File: app/searchcriteria.py (hunk 4 above)
Comment (SammyPoot) [reply app/searchcriteria.py]: Just pull the session out at the start of the method.

**Review by SammyPoot — CHANGES_REQUESTED** [review]

File: app/job_role_utils.py (hunk 1 above)
Comment (Felixim0) [reply app/job_role_utils.py]: Changed

File: app/microserviceshandler.py (hunk 2 above)
Comment (Felixim0) [reply app/microserviceshandler.py]: Adjusted

File: app/searchcriteria.py (hunk 4 above)
Comment (Felixim0) [reply app/searchcriteria.py]: ADjusted to get session from r equest

File: app/searchhandler.py (hunk 5 above)
Comment (Felixim0) [reply app/searchhandler.py]: ADjusted to get session from r equest

**Review by SammyPoot — APPROVED** [review]

**Pull request merged**